File: Python/01.ProgrammingBasics/06.Conditional-Statements-Adv-Exercise/08.OnTimefortheExam/_08.OnTimefortheExam.py
exam_hour = int(input())
exam_minute = int(input())
arrival_hour = int(input())
arrival_minute = int(input())

exam_time_in_minutes = exam_hour * 60 + exam_minute
arrival_time_in_minutes = arrival_hour * 60 + arrival_minute

difference = arrival_time_in_minutes - exam_time_in_minutes

if difference < -30:
    print("Early")
    if difference <= -60:
        hours = abs(difference) // 60
        minutes = abs(difference) % 60
        print(f"{hours}:{minutes:02d} hours before the start")
    else:
        print(f"{abs(difference)} minutes before the start")

elif -30 <= difference <= 0:
    print("On time")
    if difference != 0:
        print(f"{abs(difference)} minutes before the start")

else:
    print("Late")
    if difference >= 60:
        hours = abs(difference) // 60
        minutes = abs(difference) % 60
        print(f"{hours}:{minutes:02d} hours after the start")
    else:
        print(f"{abs(difference)} minutes after the start")

File: C++/01-Basics/03.3-ConditionalStatementsAdvЕxercise/08.OnTimefortheExam/08.OnTimefortheExam.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main() {
    int examHour, examMinute, arrivalHour, arrivalMinute;
    cin >> examHour >> examMinute >> arrivalHour >> arrivalMinute;

    int examTimeInMinutes = examHour * 60 + examMinute;
    int arrivalTimeInMinutes = arrivalHour * 60 + arrivalMinute;

    if (arrivalTimeInMinutes > examTimeInMinutes) {
        cout << "Late" << endl;
        int delay = arrivalTimeInMinutes - examTimeInMinutes;
        if (delay >= 60) {
            int hours = delay / 60;
            int minutes = delay % 60;
            cout << hours << ":" << setw(2) << setfill('0') << minutes << " hours after the start" << endl;
        }
        else {
            cout << delay << " minutes after the start" << endl;
        }
    }
    else if (arrivalTimeInMinutes >= examTimeInMinutes - 30) {
        cout << "On time" << endl;
        if (arrivalTimeInMinutes != examTimeInMinutes) {
            int minutesBeforeStart = examTimeInMinutes - arrivalTimeInMinutes;
            cout << minutesBeforeStart << " minutes before the start" << endl;
        }
    }
    else {
        cout << "Early" << endl;
        int early = examTimeInMinutes - arrivalTimeInMinutes;
        if (early >= 60) {
            int hours = early / 60;
            int minutes = early % 60;
            cout << hours << ":" << setw(2) << setfill('0') << minutes << " hours before the start" << endl;
        }
        else {
            cout << early << " minutes before the start" << endl;
        }
    }

    return 0;
}
